source/usage-examples/findOne.txt

```diff
@@ -3,3 +3,61 @@ Find a Document
 ===============
 
 .. default-domain:: mongodb
+
+You can retrieve a single document from your collection with the
+``collection.findOne()`` method. 
+
+The ``findOne()`` method takes a **context**, a **filter**, and a
+pointer to a configuration object as arguments.
+
+A context is a value of type ``context.Context``. Contexts are useful in
+concurrent server programming. A discussion of the context pattern and the
+``context`` package is outside the scope of this page, but if you are interested
+you can find more information
+`in this article on The Go Blog <https://blog.golang.org/context>`__.
+
+Your ``filter`` argument is your query to MongoDB. The ``filter`` parameter takes a query
+document that specifies the subset of your collection that you would like to
+retrieve. If you provide an empty document as your ``filter`` argument, MongoDB
+returns all the documents in your collection. The ``filter`` parameter has an empty
+interface type, but the value you pass as your ``filter`` argument must specify
+a document. For more information on documents in the MongoDB Go driver, see our
+:doc:`guide on data formats </fundamentals/data-formats>`.
```

Comment:
Thanks for the note. 

I did [some research](https://stackoverflow.com/questions/1788923/parameter-vs-argument) on the terms, and I heard it described as "functions have parameters and take arguments". So the definitions of the terms I was using were:

Argument: The value passed to a function/method.
Parameter: A value used in a function/method defined in its signature. 

I agree it is confusing, and have removed the terms from the page. Shifting the discussion away from the `findOne()` method's signature made these terms easy to remove.

I think the plan does not currently match what is in the docs-golang repo. The docs-golang repo has a [data-formats.txt page ](https://github.com/mongodb/docs-golang/blob/master/source/fundamentals/data-formats.txt) which I think corresponds to the "BSON <-> Structs" page in the plan. @terakilobyte does this sound correct to you?

Reply:
I think "filter argument" is incorrect because "argument" is the actual _value_ according to the SO page you linked, e.g. `{"name": "Alek"}`. If you wanted to use the word "filter", you could call it "argument for the filter parameter", but that is more wordy than necessary in this situation.

In general, let's avoid using SO as a definitive source of information on terminology we use and otherwise. In this case, I agree with their explanation and usage of the terms, but there is plenty of opinionated info and misinformation (sometimes because it's outdated) on that site. For drivers documentation, let's use our conventions as the first choice and then authoritative language documentation sites as secondary sources.

Reply:
Thanks for the note. Will not reference SO pages going forward, and will instead reference MongoDB terminology conventions as first choice and if no convention exists use trusted docs as second choice.